src/thread_task.rs

```diff
@@ -94,15 +102,18 @@ unsafe fn reset_task_cur(
         }
         reset_frame_idx = u32::MAX;
     }
-    if ttd.cur == 0 && ((*(c.fc).offset(first as isize)).task_thread.task_cur_prev).is_null() {
+    if ttd.cur.load(Ordering::Relaxed) == 0
+        && ((*(c.fc).offset(first as isize)).task_thread.task_cur_prev).is_null()
+    {
```

Comment:
I think there's an early return that would also need a store. We should not prematurely optimize here. If you're suggesting this for readability, I'm first worrying about correctness and doing the simplest possible change. I'm still not convinced that there aren't interleaved access to the cur field that requires re-loading it from ttd and until we can say that for sure it's safer to load and store where the original code did so.

Reply:
> I think there's an early return that would also need a store.

I'm pretty sure there aren't.  I did check.

> We should not prematurely optimize here. If you're suggesting this for readability, I'm first worrying about correctness and doing the simplest possible change. I'm still not convinced that there aren't interleaved access to the cur field that requires re-loading it from ttd and until we can say that for sure it's safer to load and store where the original code did so.

It's fine to leave this for later as an optimization, I just thought you were looking to avoid atomics where there weren't any previously.